projects/storefrontlib/src/cms-structure/services/cms-routes.service.ts

```diff
@@ -40,32 +40,40 @@ export class CmsRoutesService {
   handleCmsRoutesInGuard(
     pageContext: PageContext,
     componentTypes: string[],
-    currentUrl: string
+    currentUrl: string,
+    currentPageLabel: string
   ): boolean {
     const componentRoutes = this.cmsMapping.getRoutesForComponents(
       componentTypes
     );
     if (componentRoutes.length) {
-      if (this.updateRouting(pageContext, componentRoutes)) {
+      if (this.updateRouting(pageContext, currentPageLabel, componentRoutes)) {
         this.router.navigateByUrl(currentUrl);
         return false;
       }
     }
     return true;
   }
 
-  private updateRouting(pageContext: PageContext, routes: Route[]): boolean {
+  private updateRouting(
+    pageContext: PageContext,
+    pageLabel: string,
+    routes: Route[]
+  ): boolean {
     if (
       pageContext.type === PageType.CONTENT_PAGE &&
-      pageContext.id.startsWith('/') &&
-      pageContext.id.length > 1
+      pageLabel.startsWith('/') &&
+      pageLabel.length > 1
     ) {
       const newRoute: CmsRoute = {
-        path: pageContext.id.substr(1),
+        path: pageLabel.substr(1),
         component: PageLayoutComponent,
         children: routes,
         data: {
-          cxCmsRouteContext: pageContext,
+          cxCmsRouteContext: {
+            id: pageLabel,
```

Comment:
Do we need to put page label into context here?

Reply:
Actually, using page label causes a bug. So I used original page context again.

Reply:
After discussion, I used the page label and added page index with this page label in ngrx